api_app/analyzers_manager/observable_analyzers/spyse.py

```diff
@@ -0,0 +1,57 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+import re
+from api_app.exceptions import AnalyzerRunException
+from api_app.analyzers_manager import classes
+
+from tests.mock_utils import if_mock_connections, patch, MockResponse
+
+
+class Spyse(classes.ObservableAnalyzer):
+    base_url: str = "https://api.spyse.com/v4/data/"
+
+    def set_params(self, params):
+        self.__api_key = self._secrets["api_key_name"]
+
+    def run(self):
+        params = {"key": self.__api_key}
+        if self.observable_classification == self.ObservableTypes.DOMAIN:
+            uri = f"domain/{self.observable_name}"
+        elif self.observable_classification == self.ObservableTypes.IP:
+            uri = f"ip/{self.observable_name}"
+        elif self.observable_classification == self.ObservableTypes.GENERIC:
+            if re.match(r"^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$", self.observable_name):
+                uri = f"email/{self.observable_name}"
+            else:
```

Comment:
under the `generic`, we can also add support for the CVE endpoint. https://spyse-dev.readme.io/reference/cves